hera_cal/frf.py

```diff
@@ -220,6 +216,26 @@ def timeavg_data(self, t_avg, rephase=False, verbose=True):
         ----------
         t_avg : float
             Width of time-averaging window in seconds.
+
+        rephase : bool
+            If True, rephase data in averaging window to window-center.
+
+        data : DataContainer
```

Comment:
It seems like the only way to instantiate a VisClean object is with data. What's the point giving this optional data?

If you're looking for more flexibility, I think the right thing to do is to make a separate function that performs time averaging on datacontainers and then have this function call that one. Otherwise you have to instantiate a weird empty visclean object and then feed it external data.

Reply:
Oh the point is that you can have multiple kinds of `data` attached to the `VisClean` object, for example, there is the `self.data` which is the data loaded in, but after cleaning you also have `clean_model` and `clean_resid`, and you may want to time average those as well. If you had `VisClean.timeavg_data` hardcoded to only time-avg `self.data`, but wanted to timeavg `self.clean_model`, then you'd have to do some silly gymnastics to copy it over to `self.data`, while copy that over to `self._data` in meantime to preserve it... etc etc you get the picture. The default is `self.data`, but its nice to have it flexible for other scenarios. And adding another function to timeavg DataContainers doesn't solve that issue, I think, and frankly in my mind, isn't worth creating another entire function for, b/c its just a for loop..

Reply:
I understand now and am fine with these optional parameters for this function.

However, I still think having a separate function that operates on DataContainers is the right thing to do. As it stands (and correct me if I'm wrong), if you have a DataContainer without an associated file or HERAData/UVData object, there's no way to use this function because you have to instantiate a VisClean object to use it. 

It would be much more modular and more useful interactively if we had a stand-alone function that worked on DataContainers. The idea behind this function is that it uses all the metadata from the HERAData object correctly to perform the time averaging... and then does the for loop. It's the automatic handling of metadata that's the key value-add here. 

But if you want to supply the metadata and DataContainers yourself, you should be able to. That's the philosophy that Aaron has been pushing for modularity of analysis and I've been trying to follow. See, e.g. redcal where we have `redundantly_calibrate` which operates on DataContainers, `redcal_iteration` which wraps it and operates on HERAData objects, and `redcal_run` which wraps that and operates on files.